cxx-qt-gen/src/cxx_qt.h

```diff
@@ -73,53 +84,100 @@ class CxxQObject : public QObject
   // TODO: we need to document the existence of UpdateStateEvent for users who
   // want to create custom classes that derive from CxxQObject so that they
   // know to avoid clashes with it.
-  static const QEvent::Type UpdateStateEvent;
+  static const QEvent::Type ProcessQueueEvent;
 
 public:
   CxxQObject(QObject* parent = nullptr)
     : QObject(parent)
   {}
   virtual ~CxxQObject() = default;
 
-  void requestUpdate()
+  void requestEmitSignal(std::function<void()> signalFunctor)
   {
-    if (!m_waitingForUpdate.exchange(true, std::memory_order_relaxed))
-      QCoreApplication::postEvent(this, new QEvent(UpdateStateEvent));
+    // Lock the queue, post the event, add to the queue
+    // worst case we'll push an event that does nothing if takeQueue() is
+    // waiting on the lock
+    const std::lock_guard<std::mutex> guard(m_queueMutex);
+
+    if (!m_waitingForUpdate.exchange(true, std::memory_order_relaxed)) {
+      QCoreApplication::postEvent(this, new QEvent(ProcessQueueEvent));
+    }
+
+    m_queue.push_back(std::make_pair(QueueEvent::EmitSignal, signalFunctor));
+  }
+
+  void requestPropertyChange(std::function<void()> propertyFunctor)
```

Comment:
What purpose does this function serve? It's pretty much a duplicate of requestEmitSignal.
Also, the UpdatePropertyChange type is never really read. So is there any point in having it around?

Reply:
The only purpose was to allow for requestUpdate to track it's enum, so now that's removed, these methods are merged.